src/panels/lovelace/cards/hui-shopping-list-card.ts

```diff
@@ -0,0 +1,236 @@
+import { html, LitElement } from "@polymer/lit-element";
+import { repeat } from "lit-html/directives/repeat";
+import "@polymer/paper-checkbox/paper-checkbox.js";
+import "@polymer/paper-icon-button/paper-icon-button.js";
+import "@polymer/paper-input/paper-input.js";
+import "@polymer/paper-item/paper-icon-item.js";
+import "@polymer/paper-item/paper-item-body.js";
+import "@polymer/paper-fab/paper-fab.js";
+
+import "../../../components/ha-card.js";
+
+import { hassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { HomeAssistant } from "../../../types.js";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+import {
+  clearItems,
+  fetchItems,
+  addItem,
+  completeItem,
+  saveEdit,
+  deleteItem,
+} from "../../../data/shopping_list";
+import { TemplateResult } from "lit-html";
+
+interface Config extends LovelaceConfig {
+  title?: string;
+}
+
+export interface ShoppingListItem {
+  id: number;
+  name: string;
+  complete: boolean;
+}
+
+class HuiShoppingListCard extends hassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  public hass?: HomeAssistant;
+  protected _config?: Config;
+  private _items?: ShoppingListItem[];
+  private _unsubEvents?: () => Promise<void>;
+
+  static get properties() {
+    return {
+      _config: {},
+      _items: {},
+    };
+  }
+
+  public getCardSize(): number {
+    return (
+      (this._config ? (this._config.title ? 1 : 0) : 0) +
+      (this._items ? this._items.length : 3)
+    );
+  }
+
+  public setConfig(config: Config): void {
+    this._config = config;
+    this._items = [];
+    this._fetchData();
+  }
+
+  public connectedCallback() {
+    super.connectedCallback();
+
+    if (this.hass) {
+      this.hass.connection
+        .subscribeEvents(() => this._fetchData(), "shopping_list_updated")
+        .then(
+          (unsub: () => Promise<void>): void => {
+            this._unsubEvents = unsub;
+          }
+        );
+      this._fetchData();
+    }
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+
+    if (this._unsubEvents) {
+      this._unsubEvents();
+    }
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    return html`
+      ${this.renderStyle()}
+      <ha-card .header="${this._config.title}">
+        <div class="addRow">
+          <paper-input
+            id='addBox'
+            placeholder='${this.localize("ui.card.shopping-list.add_item")}'
+            @keydown=${this._addKeyPress}
+            no-label-float
+          ></paper-input>
+          <paper-icon-button
+            slot="item-icon"
+            icon="hass:plus"
+            @click=${this._addItem}
+          ></paper-icon-button>
+          <paper-icon-button
+            slot="item-icon"
+            icon="hass:notification-clear-all"
+            @click=${this._clearItems}
+          ></paper-icon-button>
+        </div>
+
+      ${repeat(

```

Comment:
You should add a keyFn as second arg: `item => item.id` [docs](https://polymer.github.io/lit-html/guide/writing-templates#repeatitems-keyfn-template)